mne/viz/ica.py

```diff
@@ -130,6 +130,256 @@ def plot_ica_sources(ica, inst, picks=None, exclude=None, start=None,
+    # ------------
+    if isinstance(inst, _BaseRaw):
+        # calculate segment length that gives us about 250
+        # segments and then break up continuous signal into segments
+        from ..epochs import _segment_raw
+        plot_line_at_zero = False
+        inst = _segment_raw(inst, segment_length=2., verbose=False)
+    else:
+        plot_line_at_zero = True
+
+    epochs_src = ica.get_sources(inst)
+    ica_data = epochs_src.get_data()[:, picks, :]
+    ica_data = np.swapaxes(ica_data, 0, 1)
+
+    # spectrum
+    psds, freqs = psd_multitaper(epochs_src, picks=picks, **psd_kws)
+
+    def set_title_and_labels(ax, title, xlab, ylab):
+        if title:
+            ax.set_title(title)
+        if xlab:
+            ax.set_xlabel(xlab)
+        if ylab:
+            ax.set_ylabel(ylab)
+        ax.axis('auto')
+        ax.axis('tight')
+        ax.tick_params('both', labelsize=8)
+
+    more_fig = len(picks) > 0
+    all_fig = list()
+    # the rest is component-specific
+    for idx, pick in enumerate(picks):
+        if idx > 0:
+            fig, axes = _create_properties_layout()
+
+        # spectrum
+        this_psds = psds[:, idx, :]
+        if dB:
+            this_psds = 10 * np.log10(this_psds)
+        psds_mean = this_psds.mean(axis=0)
+        diffs = this_psds - psds_mean
+        # the distribution of power for each frequency bin is highly
+        # skewed so we calculate std for values below and above average
+        # separately - this is used for fill_between shade
+        spectrum_std = [
+            [np.sqrt((d[d < 0] ** 2).mean(axis=0)) for d in diffs.T],
+            [np.sqrt((d[d > 0] ** 2).mean(axis=0)) for d in diffs.T]]
+        spectrum_std = np.array(spectrum_std) * num_std
+
+        # erp std
+        if plot_std:
+            erp = ica_data[idx].mean(axis=0)
+            diffs = ica_data[idx] - erp
+            erp_std = [
+                [np.sqrt((d[d < 0] ** 2).mean(axis=0)) for d in diffs.T],
+                [np.sqrt((d[d > 0] ** 2).mean(axis=0)) for d in diffs.T]]
+            erp_std = np.array(erp_std) * num_std
+
+        # epoch variance
+        epoch_var = np.var(ica_data[idx], axis=1)
+
+        # plotting
+        # --------
+        _plot_ica_topomap(ica, pick, show=False, axis=axes[0], **topo_kws)
+
+        # image and erp
+        plot_epochs_image(epochs_src, picks=pick, axes=axes[1:3],
+                          colorbar=False, show=False, **image_kws)
+
+        # spectrum
+        axes[3].plot(freqs, psds_mean, color='k')
+        if plot_std:
+            axes[3].fill_between(freqs, psds_mean - spectrum_std[0],
+                                 psds_mean + spectrum_std[1],
+                                 color='k', alpha=.15)
+        # epoch variance
+        axes[4].scatter(range(len(epoch_var)), epoch_var, alpha=0.5,
+                        facecolor=[0, 0, 0], lw=0)
+
+        # aesthetics
+        # ----------
+        axes[0].set_title('IC ' + str(pick))
```

Comment:
oups this is not what I meant. I meant that elsewhere we use the notation ICA001, ICA002 etc... we should try to be consistent